syntax/ebuild.vim

```diff
@@ -31,13 +31,16 @@ syn match bkshFunction	"^\s*\<\h[0-9a-zA-Z_\-\.]*\>\s*()"	skipwhite skipnl conta
 syn keyword EbuildCoreKeyword use has_version best_version use_with use_enable
 syn keyword EbuildCoreKeyword keepdir econf die einstall einfo ewarn eerror diropts
 syn keyword EbuildCoreKeyword dobin docinto dodoc doexe doheader doinfo doins
-syn keyword EbuildCoreKeyword dolib dolib.a dolib.so doman dosbin dosym emake exeinto
+syn keyword EbuildCoreKeyword doman dosbin dosym emake exeinto
 syn keyword EbuildCoreKeyword exeopts fowners fperms insinto insopts into libopts newbin
 syn keyword EbuildCoreKeyword newexe newheader newins newman newsbin has unpack into
 syn keyword EbuildCoreKeyword doinitd doconfd doenvd domo dodir ebegin eend
 syn keyword EbuildCoreKeyword newconfd newdoc newenvd newinitd newlib.a newlib.so
 syn keyword EbuildCoreKeyword hasv usev usex elog eapply eapply_user
 syn keyword EbuildCoreKeyword einstalldocs in_iuse get_libdir
+syn keyword EbuildCoreKeyword assert nonfatal docompress dostrip einfon eqawarn
+syn match EbuildCoreKeyword "\<dolib\(\.a\|\.so\)\?\_s"
```

Comment:
Could you explain why this needed to be done? In a comment, preferably.

Reply:
Added.

Reply:
Did you try escaping the dot somehow, or maybe reversing the order?

Reply:
I did yes. I tried various combinations of `dolib.a dolib.so dolib`, `dolib dolib\.a dolib\.so`, `dolib\.a dolib\.so dolib`, `'dolib.a' 'dolib.so' dolib`, and `'dolib\.a' 'dolib\.so' dolib`, as well as with double quotes instead of single quotes, and similar variations on their own lines (similarly with `newlib`) before I went with `syn match` instead of `syn keyword`.

Reply:
Your comments prompted me to do a bit more research, and I found a solution I think you'll like better than my previous `match`-based one. The reason `keyword` wasn't highlighting `dolib.*` and `newlib.*` was because '.' wasn't in the list of keyword characters. Appending '.' to `iskeyword` in `ftplugin/ebuild.vim` allowed me to restore the original `keyword` method while also matching `dolib.a` and friends.

Reply:
Great, thanks! However, this makes me wonder if we shouldn't extend that list even further ;-). I mean, right now a thing like literal `dobin/dolib` gets highlighted even though it forms a path rather than ebuild function call.

Reply:
Hmmmm, I see what you mean. I will take a look to see if I can figure something out.

Reply:
As it is right now, I don't think it's possible because `keyword` is purely keyword-based and takes precedence over `match` (which allows more complicated structures). But if I restructured the syntax file, then I think it could be done. Here's a trivial example of what I mean:

```vim
" Match paths
syn match EbuildPath "\f\+\(/\f\+\)*"
syn match EbuildCoreKeyword "^dotest" nextgroup=EbuildPath skipwhite
"... snip ...
hi def link EbuildPath String
```

Using the following test case:
```
dotest a/a
dotest dotest/a "${A}/a"
dotest dotest path/number/2 dotest
```
On the first line, `dotest` is highlighted as an `EbuildCoreKeyword` and `a/a` is highlighted as an `EbuildPath`. On the second line, the first `dotest` is highlighted as an `ECK`, but `dotest/a` is highlighted as an `EP`, and `"${A}/a"` is highlighted however the `sh.vim` file does it. On the third line, the first `dotest` is highlighted as an `ECK`, the second `dotest` is highlighted as an `EP`, as is `path/number/2` and the third `dotest`.

Note: `dotest` was defined just for experimentation.

Reply:
Just add `.` to the list of symbols that can form a keyword.